client/app/scripts/superdesk/itemList/itemList.js

```diff
@@ -0,0 +1,282 @@
+            _.each(options.notStates, function(notState) {
+                query.source.query.filtered.filter.and.push({not: {term: {state: notState}}});
+            });
+        }
+        // process state
+        if (options.states) {
+            var stateQuery = [];
+            _.each(options.states, function(state) {
+                stateQuery.push({term: {state: state}});
+            });
+            query.source.query.filtered.filter.and.push({or: stateQuery});
+        }
+        // process creation date
+        _.each(['creationDate', 'modificationDate'], function(field) {
+            if (options[field + 'Before'] || options[field + 'After']) {
+                query.source.query.filtered.filter.and.push({range: {firstcreated: {
+                    lte: options[field + 'Before'] || undefined,
+                    gte: options[field + 'After'] || undefined
+                }}});
+            }
+        });
+        // process provider, source, urgency
+        _.each(['provider', 'source', 'urgency'], function(field) {
+            if (options[field]) {
+                var directQuery = {};
+                directQuery[field] = options[field];
+                query.source.query.filtered.filter.and.push({term: directQuery});
+            }
+        });
+
+        // process search
+        var fields = {
+            headline: 'headline',
+            subject: 'subject.name',
+            keyword: 'slugline',
+            uniqueName: 'unique_name',
+            body: 'body_html'
+        };
+        var queryContent = [];
+        _.each(fields, function(dbField, field) {
+            if (options[field]) {
+                queryContent.push(dbField + ':(' + options[field] + ')');
+            }
+        });
+        if (queryContent.length) {
+            query.source.query.filtered.query = {
+                query_string: {
+                    query: queryContent.join(' '),
+                    lenient: false,
+                    default_operator: 'AND'
+                }
+            };
+        }
+        // process search
+        if (options.search) {
+            var queryContentAny = [];
+            _.each(_.values(fields), function(dbField) {
+                queryContentAny.push(dbField + ':(' + options.search + ')');
+            });
+            query.source.query.filtered.query = {
+                query_string: {
+                    query: queryContentAny.join(' '),
+                    lenient: false,
+                    default_operator: 'OR'
+                }
+            };
+        }
+
+        return query;
+    };
+    this.fetch = function(options) {
+        options = _.extend(DEFAULT_OPTIONS, options);
+        var query = getQuery(options);
+        return api(options.endpoint, options.endpointParam || undefined)
+            .query(query)
+            .then(function(result) {
+                return result;
+            });
+    };
+}])
+.provider('ItemList', function() {
```

Comment:
this is the best way to inject an instantiatable function (without instantiating, so that multiple lists can be created after injection) in my opinion.